core/src/main/java/org/elasticsearch/search/aggregations/AggregatorParsers.java

```diff
@@ -71,15 +71,14 @@ public AggregatorParsers(ParseFieldRegistry<Aggregator.Parser> aggregationParser
     /**
      * Parses the aggregation request recursively generating aggregator factories in turn.
      *
-     * @param parser    The input xcontent that will be parsed.
      * @param parseContext   The parse context.
      *
      * @return          The parsed aggregator factories.
      *
      * @throws IOException When parsing fails for unknown reasons.
      */
-    public AggregatorFactories.Builder parseAggregators(XContentParser parser, QueryParseContext parseContext) throws IOException {
-        return parseAggregators(parser, parseContext, 0);
+    public AggregatorFactories.Builder parseAggregators(QueryParseContext parseContext) throws IOException {
+        return parseAggregators(parseContext.parser(), parseContext, 0);
     }
 
     private AggregatorFactories.Builder parseAggregators(XContentParser parser, QueryParseContext parseContext, int level)
```

Comment:
Ideally yes, and we can also remove it here after https://github.com/elastic/elasticsearch/pull/17417 gets in. For the time being we need it here because later on in the method we might under certain conditions create a new `binaryParser` that will be passed down. The other option at the moment would be to store the old parser, reset the context to the new one and remember to restore the old one once the recursice call returns. Too messy IMHO and since this is a private method I think its okay.
We can get rid of this when it is easier to create new parse contexts with https://github.com/elastic/elasticsearch/pull/17417